plugins/woocommerce/includes/data-stores/class-wc-customer-data-store-session.php

```diff
@@ -78,6 +78,15 @@ public function update( &$customer ) {
 	 * @param WC_Customer $customer Customer object.
 	 */
 	public function save_to_session( $customer ) {
+		if ( ! WC()->session ) {
+			wc_doing_it_wrong(
+				__METHOD__,
+				__( 'WC_Session is not available, customer data cannot be saved to session.', 'woocommerce' ),
```

Comment:
The only other suggestion I have would be to hint at _why_ WC_Session isn't available. Based on [this](https://github.com/woocommerce/woocommerce/blob/b5eff69e16c05435ebcb30bd72dffd7864376bce/plugins/woocommerce/includes/wc-core-functions.php#L2638-L2657), I would guess that the `woocommerce_init` hook has to fire first...